goalie/mesh_seq.py

```diff
@@ -166,7 +166,7 @@ def set_meshes(self, meshes):
             :class:`firedrake.MeshGeometry`
         """
         # TODO #122: Refactor to use the set method
-        if not isinstance(meshes, Iterable):
+        if not isinstance(meshes, list):
```

Comment:
How about we allow `tuple` to cover more possible cases?
```suggestion
        if not isinstance(meshes, (list, tuple)):
```

Reply:
I committed this but I'm not sure we should allow tuples. In `adaptor` functions in demos we have `mesh_seq[i] = adapt(mesh_seq[i], metric)`, so we need a mutable object to be able to reassign meshes - I think?

Reply:
I tested this and we indeed get an error:
```
  File "/home/firedrake/goalie/demos/burgers-hessian.py", line 145, in adaptor
    mesh_seq[i] = adapt(mesh_seq[i], metric)
    ~~~~~~~~^^^
  File "/home/firedrake/goalie/goalie/mesh_seq.py", line 130, in __setitem__
    self.meshes[subinterval] = mesh
    ~~~~~~~~~~~^^^^^^^^^^^^^
TypeError: 'tuple' object does not support item assignment
```

So I reverted this. Am I missing something? Sorry if I am!